provider/ansible.py

```diff
@@ -163,19 +156,30 @@ def distro_install():
             else:
                 LOG_JOB.error("Ansible repo was required, but failed to be added.")
                 return False
-        install_status = utils_package.package_install('ansible')
+        install_status = utils_package.package_install(packages)
         if not install_status:
-            LOG_JOB.error(f"Failed to install ansbile.")
-        yum_backend.remove_repo(params["ansible_repo"])
+            LOG_JOB.error(f"Failed to install {packages}.")
+        # Remove custom dnf repo when it is no longer used
+        if params.get("ansible_repo"):
+            yum_backend.remove_repo(params["ansible_repo"])
         return install_status
 
     policy_map = {"distro_install": distro_install,
                   "python_install": python_install}
 
+    package_list = params.get_list("package_list", 'sshpass')
     try:
         path.find_command('ansible-playbook')
     except path.CmdNotFoundError:
         ansible_install_policy = params['ansible_install_policy']
-        return policy_map[ansible_install_policy]()
-    else:
-        return True
+        if ansible_install_policy == 'distro_install':
+            package_list.append('ansible')
```

Comment:
How do you think about install `ansible` as the first pkg ?

Reply:
I don't think there is a big risk regarding order of the packages, but on the other hand I think it just sounds right to install `ansible` first :slightly_smiling_face:  There is no harm to put it at the first place, so... it is done :wink: